src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcherBase.cs

```diff
@@ -1,32 +1,38 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+#nullable enable
+
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.CodeAnalysis.Razor.Workspaces
 {
-    internal class DefaultProjectSnapshotManagerDispatcher : ProjectSnapshotManagerDispatcher
+    internal abstract class ProjectSnapshotManagerDispatcherBase : ProjectSnapshotManagerDispatcher
     {
-        public override bool IsDispatcherThread
-            => Thread.CurrentThread.ManagedThreadId == ProjectSnapshotManagerTaskScheduler.Instance.ThreadId;
-
-        public override TaskScheduler DispatcherScheduler { get; } = ProjectSnapshotManagerTaskScheduler.Instance;
+        private readonly ProjectSnapshotManagerTaskScheduler _dispatcherScheduler;
 
-        internal class ProjectSnapshotManagerTaskScheduler : TaskScheduler
+        public ProjectSnapshotManagerDispatcherBase(string threadName)
         {
-            public static ProjectSnapshotManagerTaskScheduler Instance = new();
```

Comment:
This static instance was the bit that was enforcing us to use the same scheduler in VS & the server